File: accesshelper/src/main/java/com/newchar/accesshelper/BaseAccessibilityService.java
package com.newchar.accesshelper;

import android.accessibilityservice.AccessibilityService;
import android.accessibilityservice.AccessibilityServiceInfo;
import android.view.accessibility.AccessibilityEvent;

import com.newchar.accesshelper.log.LLL;

/**
 * @author newChar
 * date 2021/3/29
 * @since 服务的基础类，需要有下一层的业务服务类继承。
 * @since 迭代版本，（以及描述）
 */
public class BaseAccessibilityService extends AccessibilityService {

    private AccessManager accessManager;

    @Override
    protected void onServiceConnected() {
        super.onServiceConnected();
        LLL.w("onAccessibility", "onServiceConnected");
        accessManager = new AccessManager(this);
        accessManager.setServiceInfoChangeListener(new AccessManager.ServiceInfoChangeListener() {
            @Override
            public void onServiceInfoChange(AccessibilityServiceInfo info) {
                setServiceInfo(info);
                LLL.w("onAccessibility", "onServiceInfoChange " + info.toString());
            }
        });
        accessManager.init();
    }

    @Override
    public void onAccessibilityEvent(AccessibilityEvent event) {
        LLL.w("onAccessibility", "onAccessibilityEvent");
        if (event != null) {
//            把Event传进去，根据包名，类名过滤然后执行后边的具体判断逻辑。要执行哪些操作
            accessManager.onAccessibilityEvent(event);
        }
    }

    @Override
    public void onInterrupt() {
        if (accessManager != null) {
            accessManager.release();
            accessManager = null;
        }
    }

}

File: accesshelper/src/main/java/com/newchar/accesshelper/BaseAccessibilityService.kt
package com.newchar.accesshelper

import android.accessibilityservice.AccessibilityService
import android.accessibilityservice.AccessibilityServiceInfo
import android.content.Intent
import android.util.Log
import android.view.accessibility.AccessibilityEvent

/**
 *  @author
 *  date            2019-11-11
 *  @version
 *  @since          当前版本描述，
 *  @since          迭代版本描述
 */
open class BaseAccessibilityService : AccessibilityService() {

    private lateinit var accessManager: AccessManager

    override fun onServiceConnected() {
        super.onServiceConnected()
        Log.e("ccessibilityService", "qidongle")

        accessManager = AccessManager(this)
        accessManager.init()
        accessManager.setServiceInfoChangeListener(object :
            AccessManager.ServiceInfoChangeListener {
            override fun onServiceInfoChange(info: AccessibilityServiceInfo) {
                serviceInfo = info
            }
        })
    }

    override fun onAccessibilityEvent(event: AccessibilityEvent?) {
        event?.run {
//            把Event传进去，根据包名，类名过滤然后执行后边的具体判断罗技。要执行哪些操作
            accessManager.onAccessibilityEvent(this@BaseAccessibilityService, event)
        }

    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {

        return super.onStartCommand(intent, flags, startId)
    }

    override fun onInterrupt() {
        accessManager?.release()
    }

}